### PR TITLE
add user model

### PR DESCRIPTION
**Pull request by ggazila**
add service, dao layers for user
create AuthenticationService
add salt and hashing
change log4j2 confuguration(now levels by color)

### CODE REVIEW

File: src/main/java/mate/academy/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long userId = (Long) session.save(user);
+            transaction.commit();
+            user.setId(userId);
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.commit();
```
[the remark this answers is not in the document]
Comment (ggazila) [reply src/main/java/mate/academy/cinema/dao/impl/UserDaoImpl.java]: Oops! Fix

File: src/main/java/mate/academy/cinema/exceptions/AuthenticationException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package mate.academy.cinema.exceptions;
+
+public class AuthenticationException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (ggazila) [reply src/main/java/mate/academy/cinema/exceptions/AuthenticationException.java]: fix

File: src/main/java/mate/academy/cinema/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        StringBuilder hashPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-512");
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            LOGGER.error(e);
```
[the remark this answers is not in the document]
Comment (ggazila) [reply src/main/java/mate/academy/cinema/util/HashUtil.java]: done

File: src/main/java/mate/academy/cinema/util/HashUtil.java [hunk 4]
```diff
@@ -20,14 +20,14 @@
     public static String hashPassword(String password, byte[] salt) {
         StringBuilder hashPassword = new StringBuilder();
         try {
-            MessageDigest messageDigest = MessageDigest.getInstance("SHA-512");
+            MessageDigest messageDigest = MessageDigest.getInstance("nnn-512");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/cinema/util/HashUtil.java]: what's a point?

File: src/main/java/mate/academy/cinema/util/HashUtil.java [hunk 5]
```diff
@@ -20,14 +20,14 @@
...
             messageDigest.update(salt);
             byte[] digest = messageDigest.digest(password.getBytes());
             for (byte b : digest) {
                 hashPassword.append(String.format("%02x", b));
             }
         } catch (NoSuchAlgorithmException e) {
-            LOGGER.error(e);
+            LOGGER.error("Cannot get MessageDigest object " + e);
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/cinema/util/HashUtil.java]:
```suggestion
            LOGGER.error("Cannot get MessageDigest object ", e);
```

File: src/main/java/mate/academy/cinema/util/HashUtil.java (hunk 4 above)
Comment (ggazila) [reply src/main/java/mate/academy/cinema/util/HashUtil.java]: it was a test, fix

File: src/main/java/mate/academy/cinema/util/HashUtil.java (hunk 5 above)
Comment (ggazila) [reply src/main/java/mate/academy/cinema/util/HashUtil.java]: done

**Pull request merged**